stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -3318,16 +3318,25 @@ julia> blockdiag(sparse(2I, 3, 3), sparse(4I, 2, 2))
  ⋅  ⋅  ⋅  ⋅  4
 ```
 """
+blockdiag() = spzeros(Union{}, Int, 0, 0)
```

Comment:
I'd suggest to have
```suggestion
blockdiag() = spzeros(promote_type(), Int, 0, 0)
```
to avoid hardcoding some "non-sense" type and let the ecosystem decide. Pinging @tkf for this nasty "reduction over empty tuples" issue.

Reply:
Thanks for pointing this out, it is more sensible this way. I believe that resolves the issue, of speeding up this function while still making sure blockdiag() output remains unchanged. Would appreciate what @tkf has to say about the reduction on empty tuple issue

Reply:
I think `Union{}` makes sense (i.e., it's the identity element of `promote_type` as a monoid).  Though `promote_type()` sounds also fine.  It's just a LISPy way of saying that we want the identity of `promote_type`.

Reply:
Thanks, @tkf. I just had no idea what might be subject to potential future design changes and what is "set in stone". I'll go ahead and merge this as is then.